core/chain_indexer.go

```diff
@@ -358,6 +359,7 @@ func (c *ChainIndexer) updateLoop() {
 				} else {
 					// If processing failed, don't retry until further notification
 					c.log.Debug("Chain index processing failed", "section", section, "err", err)
+					c.verifyLastHead()
```

Comment:
Is this necessary to put the statement here? Since we will verify the validity of stored sections before processing anyway.

Reply:
You are right, it does not matter really but still I would prefer not leaving the indexer in a bad state after detecting an error.